app/Models/Concerns/HasPermissions.php

```diff
@@ -0,0 +1,221 @@
+<?php
+
+namespace App\Models\Concerns;
+
+use App\Events\User\PermissionsChanged;
+use App\Exceptions\InvalidPermissionException;
+use App\Models\Permissions\Assignment;
+use App\User;
+use Illuminate\Database\Eloquent\Relations\MorphMany;
+use Illuminate\Support\Collection;
+use VATSIMUK\Support\Auth\Facades\PermissionValidity;
+
+trait HasPermissions
+{
+    public static function bootHasPermissions()
+    {
+        static::deleting(function ($model) {
+            $model->permissions()->delete();
+        });
+    }
+
+    public function permissions(): MorphMany
+    {
+        return $this->morphMany(Assignment::class, 'related');
+    }
+
+    /**
+     * Determine if the model may perform the given permission.
+     *
+     * @param string $permission
+     *
+     * @return bool
+     */
+    public function hasPermissionTo($permission): bool
+    {
+        return $this->hasDirectPermission($permission) || $this->hasPermissionViaRole($permission);
+    }
+
+    /**
+     * Determine if the model has any of the given permissions.
+     *
+     * @param array ...$permissions
+     *
+     * @return bool
+     */
+    public function hasAnyPermission(...$permissions): bool
```

Comment:
Do you need a spread here or can you just mandate an array?

Reply:
This is taken straight from spatie's permissions. I think it gives us more options for when we want to use it